### DISCUSSION TITLE
Is there any plan to add support on mongoose library for Deno?

### DISCUSSION THREAD

**Discussion by learner00000** [denoland/deno#14053] [General]
I like Deno and I am working on a project using it. But it seems it's MongoDB driver doesn't support Mongoose library and I don't know how can I use schemas in my Deno application?

May I ask either if there is any other alternatives for Mongoose in Deno or a plan to support Mongoose or schemas in Deno soon?

**Comment by GJZwiers** [comment]
Regarding alternatives, there appears to be one in development: https://deno.land/x/deno_mongo_schema@v0.8.3

Mongoose, as far as I know, depends on Node's `crypto` module and although Deno has a Node compatibility mode, the support for Node's `crypto` is stated as 'partly' at this time: https://deno.land/manual@v1.20.1/node/compatibility_mode#node-built-in-modules. So it may not be possible to have Mongoose fully functional in Deno, it might require a port that rewrites it to use the WebCrypto API for example.

**Reply by learner00000** [reply]
Is mongo_schema as powerful as mongoose?

**Comment by erfanium** [comment]
It's off-topic but, 
As a past mongoose user, I don't recommend to use mongoose in your project. all mongodb drivers (and [Deno mongodb driver](https://github.com/denodrivers/deno_mongo) as well) has almost everything you want to create your backend.
mongoose is just an extra unnecessary abstraction layer

**Reply by learner00000** [reply]
Do you know how should I create a schema without the mongoose in a Deno project? This is my complete question nobody answered and leaded me to return and use NodeJs again:   https://stackoverflow.com/questions/71486671/is-there-anyway-working-with-schemas-in-deno?noredirect=1#comment126383549_71486671

**Reply by erfanium** [reply]
You can check this basic example
https://github.com/erfanium/deno-deploy-mongo
